src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -286,6 +295,7 @@ export const useBalanceChartData: UseBalanceChartData = args => {
   const balances = useAppSelector(state =>
     selectPortfolioCryptoBalancesByAccountId(state, accountId)
```

Comment:
we shouldn't be plumbing the balanceThreshold this deep into the useBalanceChart logic - it's complicated and fragile

we should be able to either

* write a selector to be used by the view layer that does `selectAssetIdsAboveBalanceThreshold`, and pass in that filtered list of asset ids, or
* write a selector to be used here, to that returns filtered balances instead.

Reply:
@0xdef1cafe  I was going to do exactly the first way you mentioned, but I thought if I do that, then I'm filtering out any asset which **currently** has a fiat value smaller than the threshold. so I read the whole code and decided to add the check to the `fiatBalanceAtBucket` scope to make sure if any asset had a fiat value greater than the balance threshold in any timespans, it will be shown in the chart. This probably is overkill. Is this making sense or should I rewrite this part only using `current` balances?

Reply:
yes please - that's the intention. the feature is related to assets with that current balance, e.g. a user had a bag of POOP and previously had $500 value, but they've since dumped it, have $0.01 left over, and want it filtered out.

plus - i also don't want this "view layer" logic leaking deeply down into chart computations which are already scary and hairy